src/blueprint/operators/js.rs

```diff
@@ -0,0 +1,64 @@
+use crate::blueprint::*;
+use crate::config;
+use crate::config::Field;
+use crate::lambda::Expression;
+use crate::try_fold::TryFold;
+use crate::valid::{Valid, Validator};
+
+pub struct CompileJs<'a> {
+    pub name: &'a str,
+    pub script: &'a Option<String>,
+}
+
+pub fn compile_js(inputs: CompileJs) -> Valid<Expression, String> {
+    #[cfg(not(feature = "js"))]
+    return js_disabled(inputs);
+
+    #[cfg(feature = "js")]
+    js_enabled(inputs)
+}
+
+#[cfg(not(feature = "js"))]
+fn js_disabled(_: CompileJs) -> Valid<Expression, String> {
+    Valid::fail("js is disabled or unsupported on this platform".to_string())
+}
+
+#[cfg(feature = "js")]
+fn js_enabled(inputs: CompileJs) -> Valid<Expression, String> {
+    let name = inputs.name;
+    let quickjs = rquickjs::Runtime::new().unwrap();
+    let ctx = rquickjs::Context::full(&quickjs).unwrap();
+
+    Valid::from_option(inputs.script.as_ref(), "script is required".to_string()).and_then(
+        |script| {
+            Valid::from(
+                ctx.with(|ctx| {
+                    ctx.eval::<(), &str>(script)?;
+                    Ok::<_, anyhow::Error>(())
+                })
+                .map_err(|e| crate::valid::ValidationError::new(e.to_string())),
+            )
+            .map(|_| {
+                Expression::Js(std::sync::Arc::new(crate::javascript::Runtime::new(
```

Comment:
drop ARC from here, Expression should not contain runtime info, it should be available in app_ctx